statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,703 @@
+from __future__ import division
+from math import log, pi
+import numpy as np
+from numpy import hstack, vstack
+from numpy.linalg import inv, svd
+import scipy
+import scipy.stats
+from statsmodels.tools.decorators import cache_readonly
+from statsmodels.tools.tools import chain_dot
+from statsmodels.tsa.tsatools import duplication_matrix, vec, vech, unvec
+
+import statsmodels.tsa.base.tsa_model as tsbase
+
+def mat_sqrt(_2darray):
+    """Calculates the square root of a matrix.
+
+    Parameters
+    ----------
+    _2darray : ndarray
+        A 2-dimensional ndarray representing a square matrix.
+
+    Returns
+    -------
+    result : ndarray
+        Square root of the matrix given as function argument.
+    """
+    u_, s_, v_ = svd(_2darray, full_matrices=False)
+    s_ = np.sqrt(s_)
+    return chain_dot(u_, np.diag(s_), v_)
+
+def _endog_matrices(endog_tot, diff_lags, deterministic, seasons=0):
+    """Returns different matrices needed for parameter estimation. These consist
+    of elements of the data as well as elements representing deterministic
+    terms. A tuple of consisting of these matrices is returned.
+
+    Parameters
+    ----------
+    endog_tot : ndarray (neqs x total_nobs)
+        The whole sample including the presample.
+    diff_lags : int
+        Number of lags in the VEC representation.
+    deterministic : str {"", "co", "ci", "lo", "li"}
+            "" - no deterministic terms
```

Comment:
use `"nc"` instead of empty string for no constant which is our convention in ARIMA and other models